src/main/java/com/example/EffectiveMobile/Controller/TaskController.java

```diff
@@ -1,7 +1,134 @@
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<Task> updateTask(@PathVariable Long id, @RequestBody Task task) {
+        Task existingTask = taskService.getTaskById(id);
+        if (existingTask != null) {
+            task.setId(id);
+            Task updatedTask = taskService.saveTask(task);
+            return ResponseEntity.ok(updatedTask);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteTask(@PathVariable Long id) {
+        Task existingTask = taskService.getTaskById(id);
+        if (existingTask != null) {
+            taskService.deleteTask(id);
+            return ResponseEntity.noContent().build();
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @GetMapping("/{taskId}/comments")
+    public ResponseEntity<Page<Comment>> getCommentsByTask(
+            @PathVariable Long taskId,
+            @RequestParam(defaultValue = "1") int page) {
+
+        Page<Comment> comments = commentService.getCommentsByTask(taskId, page);
+        return ResponseEntity.ok(comments);
+    }
+
+    @PostMapping("/{taskId}/comments/{commentId}")
+    public ResponseEntity<Comment> addCommentToTask(
+            @PathVariable Long taskId,
+            @RequestBody Comment comment) {
+
+        Comment createdComment = commentService.addCommentToTask(taskId, comment);
+        if (createdComment != null) {
+            return ResponseEntity.ok(createdComment);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @PutMapping("/{taskId}/comments/{commentId}")
+    public ResponseEntity<Comment> updateComment(
+            @PathVariable Long taskId,
+            @PathVariable Long commentId,
+            @RequestBody Comment comment) {
+
+        Comment updatedComment = commentService.updateComment(taskId, commentId, comment);
+        if (updatedComment != null) {
+            return ResponseEntity.ok(updatedComment);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @DeleteMapping("/{taskId}/comments/{commentId}")
+    public ResponseEntity<Void> deleteComment(
+            @PathVariable Long taskId,
+            @PathVariable Long commentId) {
+
+        commentService.deleteComment(taskId, commentId);
+        return ResponseEntity.noContent().build();
+    }
+
+    public TaskDTO TaskToDTO (Task task) {
+
+        TaskDTO taskDTO = new TaskDTO();
+        taskDTO.setTaskName(task.getTaskName());
+        taskDTO.setDescription(task.getDescription());
+        taskDTO.setPriority(task.getPriority());
+        taskDTO.setStatus(task.getStatus());
+        // Как лучше сделать ДТОху? Сделать юзер или по какому то параметру вызывать юзера?
+//        taskDTO.setAuthor(task.getAuthor().getFullName());
+//        taskDTO.setAssignee(task.getAssignee().getFullName());
```

Comment:
Usually you need Id's only. Here you don't have API to get user by ID, so I would recommend to add identifier(email). Name != identifier